app/controllers/data_types_controller.rb

```diff
@@ -0,0 +1,36 @@
+class DataTypesController < ApplicationController
+  def index
+    @data_types = DataType.all
+  end
+
+  def new
+    @data_type = DataType.new
+  end
+
+  def edit
+    @data_type = DataType.find(params[:id])
+  end
+
+  def create
+    @data_type = DataType.new(data_type_params)
+
+    @data_type.save
+    redirect_to action:"index"
+  end
+
+  def update
+    @data_type = DataType.find(params[:id])
+
+    if @data_type.update(data_type_params)
+      redirect_to action:"index"
```

Comment:
nit: Leave a space in between the colon and the quote. And in the `create` action too.